src/e2c/pdf_tools.py

```diff
@@ -4,60 +4,117 @@
 the given PDF file. Then `e2c extract` can select all words of those
 highlights with the specified color.
 """
-import pathlib
 import re
 import string
 
 import fitz
-import numpy as np
 from textblob import Word
 
-_threshold_color = 0.000001  # if two colors are different.
-_threshold_spell = 0.99  # if the word is spelled correctly.
-_threshold_intersection = 0.9  # if the intersection is large enough.
-_specials = "“" + "”"  # Some special punctuations to be removed.
+# Mac OS Preview supported colors
+PREVIEW_COLORS = {
+    "yellow": [250, 205, 90],
+    "green": [124, 200, 104],
+    "blue": [105, 176, 241],
+    "pink": [251, 92, 137],
+    "purple": [200, 133, 281],
+}
 
+PDF_ANNOT_HIGHLIGHT = 8
+SPECIAL_CHARS = "“”"
 
-def _remove_punctuation(s):
-    """Remove all kinds of punctuations in the string.
+
+def extract_words(file_path, color):
+    """Extract highlighted words with specified color in a PDF file.
 
     Args:
-        s (str): to be removed from punctuations.
+        file_path (str): target file path.
+        color (str): target color name.
 
     Returns:
-        str: with all kinds of punctuation removed.
+        list[str]: list of found words.
     """
-    table = str.maketrans({a: None for a in string.punctuation + _specials})
-    return s.translate(table)
+    if color.lower() not in PREVIEW_COLORS:
+        return []
+
+    res = set()
+    for page in fitz.open(file_path):
+        for word_tuple in sorted(
+            page.getText("words"), key=lambda w: (w[1], w[0])
+        ):
+            if word_tuple[4] in res:
+                continue
+
+            for annotation in page.annots():
+                if (
+                    annotation.type[0] == PDF_ANNOT_HIGHLIGHT
+                    and _get_color_name(annotation.colors["stroke"]) == color
```

Comment:
`annotation.colors["stroke"]` will return a list containing three numbers between 0 and 1. We should convert it. Will have a look today.

Reply:
This has been covered in `_get_color_name `